src/libtls/tls_server.c

```diff
@@ -172,8 +184,19 @@ static bool select_suite_and_key(private_tls_server_t *this,
 			 this->server);
 		return FALSE;
 	}
-	this->suite = this->crypto->select_cipher_suite(this->crypto,
-											suites, count, key->get_type(key));
+
+	if (this->tls->get_version_max(this->tls) >= TLS_1_3)
+	{
+		this->suite = this->crypto->select_cipher_suite(this->crypto, suites,
+												  		count, KEY_ANY);
+	}
+	else
+	{
+		this->suite = this->crypto->select_cipher_suite(this->crypto, suites,
+												  		count,
+												  		key->get_type(key));
+	}
+
 	if (!this->suite)
 	{	/* no match for this key, try to find another type */
 		if (key->get_type(key) == KEY_ECDSA)
```

Comment:
What do you mean?

Reply:
I think I understand the idea behind this condition now: If this->suite is not yet set, then try to find another CS with switched key type. RSA instead of ECDSA and ECDSA instead of RSA. Correct?

Reply:
see #15.